conans/client/generators/msbuild.py

```diff
@@ -0,0 +1,198 @@
+            <Conan{name}CompilerFlags>{compiler_flags}</Conan{name}CompilerFlags>
+            <Conan{name}LinkerFlags>{linker_flags}</Conan{name}LinkerFlags>
+            <Conan{name}PreprocessorDefinitions>{definitions}</Conan{name}PreprocessorDefinitions>
+            <Conan{name}IncludeDirectories>{include_dirs}</Conan{name}IncludeDirectories>
+            <Conan{name}ResourceDirectories>{res_dirs}</Conan{name}ResourceDirectories>
+            <Conan{name}LibraryDirectories>{lib_dirs}</Conan{name}LibraryDirectories>
+            <Conan{name}BinaryDirectories>{bin_dirs}</Conan{name}BinaryDirectories>
+            <Conan{name}Libraries>{libs}</Conan{name}Libraries>
+            <Conan{name}SystemDeps>{system_libs}</Conan{name}SystemDeps>
+          </PropertyGroup>
+          <PropertyGroup>
+            <LocalDebuggerEnvironment>PATH=%PATH%;$(Conan{name}BinaryDirectories)</LocalDebuggerEnvironment>
+            <DebuggerFlavor>WindowsLocalDebugger</DebuggerFlavor>
+          </PropertyGroup>
+          <ItemDefinitionGroup>
+            <ClCompile>
+              <AdditionalIncludeDirectories>$(Conan{name}IncludeDirectories)%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+              <PreprocessorDefinitions>$(Conan{name}PreprocessorDefinitions)%(PreprocessorDefinitions)</PreprocessorDefinitions>
+              <AdditionalOptions>$(Conan{name}CompilerFlags) %(AdditionalOptions)</AdditionalOptions>
+            </ClCompile>
+            <Link>
+              <AdditionalLibraryDirectories>$(Conan{name}LibraryDirectories)%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
+              <AdditionalDependencies>$(Conan{name}Libraries)%(AdditionalDependencies)</AdditionalDependencies>
+              <AdditionalDependencies>$(Conan{name}SystemDeps)%(AdditionalDependencies)</AdditionalDependencies>
+              <AdditionalOptions>$(Conan{name}LinkerFlags) %(AdditionalOptions)</AdditionalOptions>
+            </Link>
+            <Midl>
+              <AdditionalIncludeDirectories>$(Conan{name}IncludeDirectories)%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+            </Midl>
+            <ResourceCompile>
+              <AdditionalIncludeDirectories>$(Conan{name}IncludeDirectories)%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+              <PreprocessorDefinitions>$(Conan{name}PreprocessorDefinitions)%(PreprocessorDefinitions)</PreprocessorDefinitions>
+              <AdditionalOptions>$(Conan{name}CompilerFlags) %(AdditionalOptions)</AdditionalOptions>
+            </ResourceCompile>
+          </ItemDefinitionGroup>
+        </Project>
+        """)
+
+    @property
+    def filename(self):
+        return None
+
+    @ staticmethod
+    def _name_condition(settings):
+        toolset = msvs_toolset(settings)
+        if toolset is None:
+            raise ConanException("Undefined Visual Studio version %s" %
+                                 settings.get_safe("compiler.version"))
```

Comment:
If we are already checkint it is `compiler=Visual Studio`, then we don't need to check anything here

```suggestion
```